### ISSUE TITLE
Split Pane feature - feels too primitive and needs overhaul

### ISSUE THREAD

**Issue by ADTC** [warpdotdev/Warp#5303]
### Dupe Check

- [X] I have [searched Warp feature requests](https://github.com/warpdotdev/warp/issues?q=is%3Aissue+label%3A%22Feature+requests%22) and there are no duplicates

### Describe the solution you'd like?

Split Pane feature seems very primitive right now, and it leaves a lot to be desired.

* No way to toggle panes (swap left pane with right or upper pane with lower pane).
* No way to change pane configuration (change left/right to up/down or vice/versa)
* No way to drag panes around to put them in a different side or position in the current tab.
* No way to drag and drop another tab to a side of current tab's contents to move it into a pane inside current tab.
* No way to un-split a pane into its own tab.
* Cannot double-click the pane separator to reset it to a 50/50 split.
* The option to maximize/minimize pane is inconveniently hidden in a menu with _only one option_. I'd prefer if the option is its own button for easy access. (There's a keyboard shortcut, which alleviates the pain a bit.)

So much of this is possible and intuitive in VS Code which is my point of reference for how well-done the Split Pane feature should be.

A separate but relevant issue is: There's no ability to control word wrap of the output. It will be nice to change _"pane-width-only wrapping"_ to _"line-width-prioritized wrapping"_ instead and allow horizontal scrolling. Basically, output can wrap to the window/pane if wider, but if it's too narrow, it should have a minimum line length of say 100 characters and have horizontal scrolling.

It's relevant as panes in a split-pane setup can be narrow and it would be nice to see the wrapping not strictly follow the pane width. (The wrapping mode should be configurable, including the minimum line width.)

### How important is this feature to you?

4

### Warp Internal (ignore) - linear-label:39cc6478-1249-4ee7-950b-c428edfeecd1

None

**Comment by dannyneira** [comment]
Thanks for this feature request @ADTC! Warp does have many of the pane management features you mentioned. Please see more details on them in our [Split Pane docs](https://docs.warp.dev/features/windows/split-panes) as well as the demo I've attached below:

https://github.com/user-attachments/assets/fc3bf0f5-2e39-4a10-89db-307db8dd1400

To anyone else interested in this feature, please add a 👍 to the original post at the top to signal that you want this feature, and subscribe if you'd like to be notified.

**Comment by ADTC** [comment]
LOL, coincidentally I just discovered this by a sheer accident of the mouse. I had not thought of dragging the title bar of the pane. I should probably RTFM more.

However, it seems I cannot drag from the main tab bar into the current tab to move a tab into a new pane. At least it should be possible if the dragged tab is not a tab with split panes already. (The workaround is to split the tab I want to move, then drag the split pane out of there, into the tab where it should go.)

**Comment by dannyneira** [comment]
Yes, that's right it's doesnt work from tab to pane currently. There is also a related request to drag tabs/panes into separate windows here. https://github.com/warpdotdev/Warp/issues/466
